### ISSUE TITLE
Add "--depth 1" argument for faster git cloning

### ISSUE THREAD

**Issue by yaskhan** [rust-lang/cargo#1956]
http://stackoverflow.com/questions/26957237/faster-git-clone

**Comment by alexcrichton** [comment]
This is a duplicate of https://github.com/rust-lang/cargo/issues/1171, and unfortunately that's blocked on upstream support in libgit2, but #1171 should receive status updates for this!
